middleware/apimiddleware.go

```diff
@@ -479,6 +480,24 @@ func AddressPostCtx(next http.Handler) http.Handler {
 	})
 }
 
+// DenySpecificAddr https://github.com/decred/dcrdata/issues/358
+func DenySpecificAddr(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		address := GetAddressCtx(r)
+		params := &chaincfg.MainNetParams
```

Comment:
This has to work for all nets.

Need `DenyAddress(address string) func(http.Handler) http.Handler` that captures the address in a closure (the actual middleware).

Reply:
To clarify, `chaincfg.MainNetParams` is going to generate the address DsQxuVRvS4eaJ42dhQEsCXauMWjvopWgrVg .  But testnet has a different address.

Reply:
I've renamed this from deny to "DsQxuVRvS4eaJ42dhQEsCXauMWjvopWgrVg" to deny "zero hash P2PHK address" to emphasize that this needs to work on all nets (mainnet/testnetX/simnet).